server/src/main/webapp/WEB-INF/rails/webpack/models/new-environments/environments_apis.ts

```diff
@@ -45,11 +45,23 @@ export class EnvironmentsAPIs {
   static patch(name: string, payload: EnvironmentPatchJson) {
     return ApiRequestBuilder.PATCH(SparkRoutes.apiAdminEnvironmentsPath(name), this.LATEST_API_VERSION_HEADER, {payload});
   }
+
+  static updateAgentAssociation(envName: string, agentUuids: string[]) {
+    return ApiRequestBuilder.PUT(SparkRoutes.apiAdminInternalEnvironmentsPath(envName), this.LATEST_API_VERSION_HEADER, {
```

Comment:
Can we use public API instead of internal API? or this is a conscious decision we have made?

Reply:
There is no public API which can be used to modify single env-multiple agent association. Hence had to create an internal API for usage in SPAs.